doc/glossary.rst

```diff
@@ -1854,20 +1854,60 @@ See concept :term:`sample property`.
+        dataset of observations with different exposure durations per
+        individual (see
+        :ref:`sphx_glr_auto_examples_linear_model_plot_poisson_regression_non_normal_loss.py`
+        and
+        :ref:`sphx_glr_auto_examples_linear_model_plot_tweedie_regression_insurance_claims.py`).
+
+        Third-party libraries can also use `sample_weight`-compatible
+        estimators as building blocks to reduce a specific statistical task
+        into a weighted regression or classification task. For instance sample
+        weights can be constructed to adjust a time-to-event model for
+        censoring in a predictive survival analysis setting. In causal
+        inference, it is possible to reduce a conditional average treatment
+        effect estimation task to a weighted regression task under some
+        assumptions. Sample weights can also be used to mitigate
+        fairness-related harms based on a given quantitative definition of
+        fairness.
+
+        Some model hyper-parameters are expressed in terms of a discrete number
+        of samples in a region of the feature space. When fitting with sample
+        weights, a count of samples is often automatically converted to a sum
+        of their weights as is the case  for `min_samples` in
+        :class:`cluster.DBSCAN`, for instance. However, this is not always the
+        case. In particular, the ``min_samples_leaf`` parameter in
+        :class:`ensemble.RandomForestClassifier` does not take weights into
+        account. One should instead pass `min_weight_fraction_leaf` to
+        :class:`ensemble.RandomForestClassifier` to specify the minimum sum of
+        weights of samples in a leaf.
+
+        In classification, weights can also be specified for all samples
+        belonging to a given target class with the :term:`class_weight`
+        estimator :term:`parameter`. If both ``sample_weight`` and
+        ``class_weight`` are provided, the final weight assigned to a sample is
+        the product of the two.
+
+        `sample_weight` can be both an argument of the estimator's `fit` method
+        for model training or a parameter of a :term:`scorer` for model
+        evaluation.
+
+        At the time of writing, not all scikit-learn estimators correctly
+        implement the weight-repetition equivalence property. The `#16298 meta
+        issue <https://github.com/scikit-learn/scikit-learn/issues/16298>`_
+        tracks ongoing work to detect and fix remaining discrepancies.
+
+        Furthermore, some estimators have a stochastic fit method. For
+        instance, :class:`cluster.KMeans` depends on a random initialization,
+        bagging models randomly resample from the training data, etc. In this
+        case, the sample weight-repetition equivalence property described above
+        does not hold exactly. However, it should hold at least in expectation
+        over the randomness of the fitting procedure.
```

Comment:
Note to reviewers: I think the equivalence should even hold in distribution rather than just in expectation, but I am not 100% sure this can always be enforced in scikit-learn (yet).